File: backend/app/api/routes/topics.py
from typing import List
from fastapi import APIRouter
from app.schemas.conversation import TopicItem, TopicsGroupResponse

router = APIRouter(prefix="/api/topics", tags=["Topics"])

TOPIC_CATALOG = [
    TopicsGroupResponse(
        category="Everyday",
        topics=[
            TopicItem(
                id="daily_life",
                title="Daily Life & Routines",
                subtitle="Morning habits, daily schedules, and weekend plans",
                category="Everyday",
                icon="wb_sunny",
                starter_prompt="Tell me, how does a typical morning start for you?"
            ),
            TopicItem(
                id="food_dining",
                title="Food & Dining Out",
                subtitle="Favorite dishes, cooking at home, and restaurant chats",
                category="Everyday",
                icon="restaurant",
                starter_prompt="What is your absolute favorite meal to eat or cook?"
            ),
            TopicItem(
                id="shopping_culture",
                title="Shopping & Markets",
                subtitle="Bargaining, finding clothes, and shopping online",
                category="Everyday",
                icon="shopping_bag",
                starter_prompt="Do you prefer shopping in physical stores or online? Why?"
            ),
            TopicItem(
                id="hobbies_passions",
                title="Hobbies & Free Time",
                subtitle="Music, sports, gaming, and relaxing after work",
                category="Everyday",
                icon="sports_esports",
                starter_prompt="What kind of activities help you unwind after a busy day?"
            ),
            TopicItem(
                id="family_friends",
                title="Friends & Relationships",
                subtitle="Catching up with old friends and family gatherings",
                category="Everyday",
                icon="groups",
                starter_prompt="How do you usually spend time with your closest friends?"
            ),
            TopicItem(
                id="travel_adventures",
                title="Travel & Exploration",
                subtitle="Vacations, discovering new cities, and travel tips",
                category="Everyday",
                icon="flight_takeoff",
                starter_prompt="If you could book a flight anywhere tomorrow, where would you go?"
            ),
        ]
    ),
    TopicsGroupResponse(
        category="Professional",
        topics=[
            TopicItem(
                id="job_interview",
                title="Job Interview Practice",
                subtitle="Role pitching, answering tough questions, and strengths",
                category="Professional",
                icon="work_outline",
                starter_prompt="Let's start the interview! Could you please introduce yourself and walk me through your background?"
            ),
            TopicItem(
                id="introducing_yourself",
                title="Professional Introduction",
                subtitle="Elevator pitch, sharing your expertise and current role",
                category="Professional",
                icon="badge",
                starter_prompt="How would you introduce your current role and goals in 30 seconds?"
            ),
            TopicItem(
                id="teamwork_collaboration",
                title="Teamwork & Collaboration",
                subtitle="Handling disagreements, team projects, and leadership",
                category="Professional",
                icon="handshake",
                starter_prompt="Can you describe a time when you worked on a challenging project with a team?"
            ),
            TopicItem(
                id="workplace_communication",
                title="Workplace Communication",
                subtitle="Writing emails, giving feedback, and leading meetings",
                category="Professional",
                icon="forum",
                starter_prompt="What do you find most important when communicating with colleagues remotely?"
            ),
        ]
    ),
    TopicsGroupResponse(
        category="Advanced",
        topics=[
            TopicItem(
                id="ai_technology",
                title="Artificial Intelligence & Tech",
                subtitle="Ethical dilemmas, automated future, and breakthroughs",
                category="Advanced",
                icon="smart_toy",
                starter_prompt="Do you think artificial intelligence will create more opportunities or challenges for humanity?"
            ),
            TopicItem(
                id="climate_sustainability",
                title="Climate & Sustainability",
                subtitle="Renewable energy, urban ecology, and individual action",
                category="Advanced",
                icon="eco",
                starter_prompt="In your opinion, what is the most pressing environmental issue our generation faces?"
            ),
            TopicItem(
                id="future_of_work",
                title="The Future of Work",
                subtitle="Remote culture, gig economy, and four-day workweeks",
                category="Advanced",
                icon="psychology",
                starter_prompt="How do you see the traditional 9-to-5 workplace evolving over the next decade?"
            ),
            TopicItem(
                id="society_culture",
                title="Culture & Modern Society",
                subtitle="Globalization, social media impact, and community",
                category="Advanced",
                icon="public",
                starter_prompt="How has modern social media changed the way we form relationships?"
            ),
        ]
    ),
    TopicsGroupResponse(
        category="Free Conversation",
        topics=[
            TopicItem(
                id="free_talk",
                title="Open Conversation",
                subtitle="Talk about anything on your mind with no set agenda",
                category="Free Conversation",
                icon="record_voice_over",
                starter_prompt="Hello! I'm Echo. What's on your mind today? We can talk about anything you'd like."
            )
        ]
    )
]


@router.get("", response_model=List[TopicsGroupResponse])
def get_all_topics():
    """Returns categorized list of speaking practice topics."""
    return TOPIC_CATALOG

File: mobile/constants/topics.ts
import { Topic } from '../types';

export const PRACTICE_TOPICS: Topic[] = [
  // Everyday
  {
    id: 'daily_life',
    title: 'Daily Life & Routines',
    subtitle: 'Morning habits, schedules, and everyday routines',
    category: 'Everyday',
    icon: 'wb_sunny',
    starter_prompt: "Tell me, how does a typical morning start for you?"
  },
  {
    id: 'food_dining',
    title: 'Food & Cooking',
    subtitle: 'Favorite meals, dining out, and recipes',
    category: 'Everyday',
    icon: 'restaurant',
    starter_prompt: "What is your absolute favorite dish to eat or cook?"
  },
  {
    id: 'travel_cities',
    title: 'Travel & Exploration',
    subtitle: 'Vacations, discovering new cities, and packing',
    category: 'Everyday',
    icon: 'flight_takeoff',
    starter_prompt: "If you could book a flight anywhere in the world right now, where would you go?"
  },
  {
    id: 'coffee_culture',
    title: 'Coffee & Informal Chats',
    subtitle: 'Cafe conversations, relaxing, and small talk',
    category: 'Everyday',
    icon: 'coffee',
    starter_prompt: "Do you like spending time in coffee shops? What's your usual order?"
  },
  {
    id: 'hobbies_leisure',
    title: 'Hobbies & Passions',
    subtitle: 'Music, gaming, outdoor activities, and books',
    category: 'Everyday',
    icon: 'sports_esports',
    starter_prompt: "What hobbies or activities do you enjoy doing during your free time?"
  },

  // Professional
  {
    id: 'job_interview',
    title: 'Job Interview Practice',
    subtitle: 'Role pitching, answering difficult questions, and career goals',
    category: 'Professional',
    icon: 'work_outline',
    starter_prompt: "Welcome to the interview! Can you introduce yourself and walk me through your background?"
  },
  {
    id: 'teamwork_projects',
    title: 'Teamwork & Projects',
    subtitle: 'Overcoming challenges and collaborating with peers',
    category: 'Professional',
    icon: 'groups',
    starter_prompt: "Can you tell me about a time you worked on a challenging project with a team?"
  },
  {
    id: 'workplace_communication',
    title: 'Workplace Communication',
    subtitle: 'Giving constructive feedback and leading discussions',
    category: 'Professional',
    icon: 'forum',
    starter_prompt: "How do you prefer communicating important updates to your colleagues?"
  },

  // Advanced
  {
    id: 'ai_technology',
    title: 'AI & The Future of Tech',
    subtitle: 'Automation, ethics, and future societal impacts',
    category: 'Advanced',
    icon: 'smart_toy',
    starter_prompt: "Do you believe artificial intelligence will create more opportunities or challenges for future generations?"
  },
  {
    id: 'climate_sustainability',
    title: 'Climate & Sustainability',
    subtitle: 'Renewable energy, urban ecology, and environment',
    category: 'Advanced',
    icon: 'eco',
    starter_prompt: "In your perspective, what is the most important environmental issue we face today?"
  },

  // Free Conversation
  {
    id: 'free_talk',
    title: 'Open Conversation',
    subtitle: 'Talk about anything on your mind with no agenda',
    category: 'Free Conversation',
    icon: 'record_voice_over',
    starter_prompt: "Hello! I'm Echo. What is on your mind today? Feel free to speak about whatever you like."
  }
];
